compiler/rustc_const_eval/src/const_eval/error.rs

```diff
@@ -162,15 +163,18 @@ where
             // Report as hard error.
             let (our_span, frames) = get_span_and_frames();
             let span = span.unwrap_or(our_span);
-            let err = mk(span, frames);
-            let mut err = tcx.sess.create_err(err);
+            let err = tcx.sess.create_err(Spanned { span, node: InterpErrorExt(error) });
 
-            let msg = error.diagnostic_message();
-            error.add_args(&tcx.sess.parse_sess.span_diagnostic, &mut err);
+            let Some((mut err, handler)) = err.into_diagnostic() else {
+                panic!("did not emit diag");
+            };
+
+            for frame in frames {
+                err.eager_subdiagnostic(handler, frame);
+            }
```

Comment:
the handler is at `&tcx.sess.parse_sess.span_diagnostic` so you don't need to use `into_diagnostic` here and you should call methods on the builder instead. and then you can call the builder's `emit` method.